romancal/resample/resample_utils.py

```diff
@@ -401,3 +416,19 @@ def resample_range(data_shape, bbox=None):
         ymax = min(data_shape[0] - 1, int(y2 + 0.5))
 
     return xmin, xmax, ymin, ymax
+
+
+def add_var_sky_array(input_models: ModelLibrary):
+    input_models = (
+        input_models
+        if isinstance(input_models, ModelLibrary)
+        else ModelLibrary([input_models])
+    )
+    with input_models:
+        ref_img = input_models.borrow(index=0)
+        var_sky = np.zeros_like(ref_img.data)
+        input_models.shelve(model=ref_img, index=0)
+        for i, img in enumerate(input_models):
+            var_sky += img.var_rnoise + img.var_poisson / img.data * np.median(img.data)
+            img["var_sky"] = var_sky
+            input_models.shelve(img, i, modify=True)
```

Comment:
I think this pattern will result in a different `var_sky` if this step is run with an on_disk vs in_memory ModelLibrary.
Let's say we have 3 models in the library. The first time through this loop `var_sky` will be set as in the code above and then assigned to the first model under the `var_sky` attribute. The `shelve` will write out this array if the library is "on_disk" freezing the contents of `var_sky` so that for the first model it's `var_sky` will only include contributions from the first model. If the library is in memory the `var_sky` array for model 1 will stay in memory and be updated the 2nd (and 3rd) times through the loop. So for an in memory library the `var_sky` for each model will be determined by all models but for an on disk library the `var_sky` will only be determined by that model (and any previous).
If the goal is to have `var_sky` be a combination of all models the assignment to `img["var_sky"]` could be broken into a separate loop or alternatively just kept in memory (since it would be the same for all models).

Reply:
Thanks for the reminder that we also have an "on disk" mode. ;)
I will reformat this to account for that.

Reply:
Actually, `add_var_sky_array()` is meant to add the `var_sky` array to the L2 images (in memory) so that they can be combined later on by drizzle using `weight_type="ivsky"`.